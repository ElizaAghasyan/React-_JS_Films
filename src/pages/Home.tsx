import React from "react";
import Header from "../components/header/Header";

const styles = require("./Home.module.scss");

const Home = () => {
    return (
        <div className={styles.wrapper}>
            <Header />
        </div>
    );
}

export default Home;
