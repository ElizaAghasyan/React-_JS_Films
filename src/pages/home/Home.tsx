import React, { useState, useEffect } from "react";
import Banner from "../../components/banner/Banner";
import {getMovieList, getTrending, movieType} from "../../config/movieApi";
import Navigation from "../../components/navigation/Navigation";
import MovieList from "../../components/movieList/MovieList";
import Loading from "../../components/loading/Loading";

const styles = require("./Home.module.scss");

export type StateProperties = {
    id: number | string;
    title: string;
    overview: string;
    poster_path: string;
    vote_average: number;
    backdrop_path: string;
    vote_count: number;
    name: string;
}

const Home = () => {
    const [movie, setMovie] = useState<StateProperties[]>([]);
    const [page, setPage] = useState(1);

    useEffect(() => {
        const getPopular = async () => {
            const params = {page: 1}
            const response = await getMovieList(movieType.popular, {params})
                .catch((err: string) => {
                    console.log(err)
                })
            setMovie(response?.data.results.slice(0,12))
        }
        getPopular()
    }, []);

    const loadMore = async () => {
        const params = {
            page: page + 1
        };
        let  response = await getMovieList(movieType.upcoming, {params});
        setMovie([...movie, ...response.data.results]);
        setPage(page + 1)
    }

    const handleClick= (e: any) => {
        const getData = async () => {
            let response;
            let li = e.target.innerText
            const params = {}
            switch (li) {
                case 'Top Rated':
                    response = await getMovieList(movieType.top_rated, {params});
                    break;
                case 'Trending':
                    response = await getTrending(movieType.trending, {params});
                    break;
                case 'Coming Soon':
                    response = await getMovieList(movieType.upcoming, {params: {page: 2}});
                    break;
                default:
                    response = await getMovieList(movieType.popular, {params: {page: 2}});
                    break;
            }
            setMovie(response.data.results.slice(0, 16));
        }
        getData();
    }

    return (
        <div className={styles.wrapper}>
            <Banner />
            <Navigation click={handleClick} />
            <MovieList item={movie} />
            <div onClick={loadMore}>
                <Loading  />
            </div>
        </div>
    );
}

export default Home;
