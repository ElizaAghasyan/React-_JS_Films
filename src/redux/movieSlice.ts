import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
import serverApi, {movieType} from "../config/movieApi";

const initialState = {
    movies: {},
}

interface MovieType  {
    id: number;
    title: string;
    overview: string;
    poster_path: string;
    vote_average: number;
    backdrop_path: string;
    vote_count: number;
    name: string;
}


export const fetchAsyncMovies = createAsyncThunk('movies/fetchAsyncMovies', async () => {
    const params = {page: 1}
    let response = await serverApi.getMovieList(movieType.popular, {params})
        .catch(err => {
            console.log(err)
        })
    let result = await response?.data.results[Math.floor(Math.random() * response.data.results.length - 1)]
    return result;
});


export const movieSlice = createSlice({
    name: 'movies',
    initialState,
    reducers: {
        addMovies: (state, {payload}) => {
            state.movies = payload
        }
    },
    extraReducers: (builder) => {
        builder.addCase(fetchAsyncMovies.pending, (state, action) => {
            console.log("Pending...")
        })
        builder.addCase(fetchAsyncMovies.fulfilled, (state: {}, action) => {
            return {
                ...state, movies: action.payload
            }
        })
        builder.addCase(fetchAsyncMovies.rejected, (state, action) => {
            console.log("Pending...")
        })
    }
});

export const {addMovies} = movieSlice.actions;

export const getBannerMovie = (state: { movies: { movies: MovieType }}) => state.movies.movies;


export default movieSlice.reducer;
