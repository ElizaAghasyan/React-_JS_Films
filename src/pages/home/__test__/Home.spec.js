import Home from "../Home";
import renderer from "react-test-renderer";
import {store} from "../../../redux/store";
import {Provider} from "react-redux";

describe('Home Component', () => {

    it('Home snapshot test', () => {
        const component = renderer.create(<Provider store={store}><Home /></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
