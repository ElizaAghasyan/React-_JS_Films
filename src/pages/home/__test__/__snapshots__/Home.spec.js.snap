// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Home snapshot test 1`] = `
<div
  className="wrapper"
>
  <div
    className="header_content"
  >
    <h1>
      Films
    </h1>
    <div>
      <input
        onChange={[Function]}
        placeholder="Search Films"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchOutlinedIcon"
        focusable="false"
        style={
          Object {
            "color": "#fff",
            "fontSize": "3em",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
  <div
    className="banner"
    style={
      Object {
        "backgroundImage": "url(null)",
        "width": "100%",
      }
    }
  >
    <div
      className="banner_content"
    >
      <div
        className="banner_content_info"
      >
        <h2 />
        <ul>
          <li>
            Adventure
          </li>
          <li>
            Fantasy
          </li>
          <li>
            Family
          </li>
          <li>
            Comedy
          </li>
        </ul>
        <div
          className="stars"
        >
          <span
            className=""
          >
            <span
              aria-hidden="true"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#3bb3df",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "height": 20,
                  "transition": "",
                  "width": 20,
                }
              }
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 24 24"
                width={20}
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              aria-hidden="true"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#3bb3df",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "height": 20,
                  "transition": "",
                  "width": 20,
                }
              }
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 24 24"
                width={20}
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              aria-hidden="true"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#3bb3df",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "height": 20,
                  "transition": "",
                  "width": 20,
                }
              }
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 24 24"
                width={20}
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              aria-hidden="true"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#3bb3df",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "height": 20,
                  "transition": "",
                  "width": 20,
                }
              }
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 24 24"
                width={20}
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              aria-hidden="true"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#3bb3df",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "height": 20,
                  "transition": "",
                  "width": 20,
                }
              }
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 24 24"
                width={20}
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <button />
        </div>
      </div>
      <div
        className="buttons"
      >
        <button
          onClick={[Function]}
        >
          Watch Now
        </button>
        <button
          onClick={[Function]}
        >
          View Info
        </button>
      </div>
      <div
        className="details"
      />
      <div
        className="modal"
      >
        <div
          className="modal__content"
        >
          <iframe
            frameBorder="0"
            height="500px"
            title="trailer"
            width="100%"
          />
        </div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-close-1 css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </div>
  </div>
  <section
    className="content"
  >
    <div
      className="button_content"
    >
      <ul>
        <li
          className="buttons"
          onClick={[Function]}
          value="Popular"
        >
          Popular
        </li>
        <li
          className="buttons"
          onClick={[Function]}
          value="Trending"
        >
          Trending
        </li>
        <li
          className="buttons"
          onClick={[Function]}
          value="Top Rated"
        >
          Top Rated
        </li>
        <li
          className="buttons"
          onClick={[Function]}
          value="Coming Soon"
        >
          Coming Soon
        </li>
      </ul>
      <div
        className="icons"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DashboardIcon"
          focusable="false"
          style={
            Object {
              "color": "#d6d7d7",
              "fontSize": "2.8em",
              "margin": "-1px 5px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DashboardIcon"
          focusable="false"
          style={
            Object {
              "color": "#3bb3df",
              "fontSize": "2.5em",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
          />
        </svg>
      </div>
    </div>
  </section>
  <div
    className="cards"
  />
  <div
    onClick={[Function]}
  >
    <div
      className="dots"
    >
      <div
        className="spinner"
      >
        <div
          className="dot1"
        >
           
        </div>
        <div
          className="dot2"
        >
           
        </div>
        <div
          className="dot3"
        >
           
        </div>
      </div>
      <p>
        LOADING
      </p>
    </div>
  </div>
  <div
    className="footer"
  >
    <h4>
      Films
    </h4>
    <p>
      Copyright Â© 2018 FILMS. YOUR NAME
    </p>
  </div>
</div>
`;
